### PR TITLE
[Issue #48057] Removing usages of ToXContentParams.INCLUDE_TYPE

### LINKED ISSUE #48057: [ML] Remove ToXContentParams.INCLUDE_TYPE
Given we now have `INCLUDE_TYPE` and `FOR_INTERNAL_STORAGE` it seems redundant to keep both. `FOR_INTERNAL_STORAGE` can be used in all cases where we are using `INCLUDE_TYPE` plus more. It will simplify things to just keep the one.

### PR DESCRIPTION
**Pull request by rsarawgi**
Removing the option of ToXContentParams.INCLUDE_TYPE and replacing them with ToXContentParams.FOR_INTERNAL_STORAGE 
Closes #48057

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @rsarawgi, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/elasticsearch/pull/48165.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java [hunk 1]
```diff
@@ -63,7 +63,6 @@
 
     static {
         Map<String, String> modifiable = new HashMap<>();
-        modifiable.put(ToXContentParams.INCLUDE_TYPE, "true");
         modifiable.put(ToXContentParams.FOR_INTERNAL_STORAGE, "true");
         TO_XCONTENT_PARAMS = Collections.unmodifiableMap(modifiable);
     }
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java]: Now this only has one entry this whole `static` block can be removed and the declaration above changed to simply:

```
private static final Map<String, String> TO_XCONTENT_PARAMS = Collections.singletonMap(ToXContentParams.FOR_INTERNAL_STORAGE, "true");
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ToXContentParams.java [hunk 2]
```diff
@@ -14,16 +14,10 @@
     /**
      * Parameter to indicate whether we are serialising to X Content for
      * internal storage. Certain fields need to be persisted but should
-     * not be visible everywhere.
+     * not be visible everywhere
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ToXContentParams.java]: Why remove the full stop at the end of the sentence?

**Review by droberts195 — COMMENTED** [review]
Thanks for doing this @rsarawgi.  I left a couple of minor comments about tidying it up slightly.  Also please see the automated comment above about tying your signature on the CLA to your GitHub account.  Unfortunately we cannot accept contributions unless all the legal boxes are ticked.

**Comment by droberts195** [comment]
Jenkins test this

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ToXContentParams.java (hunk 2 above)
Comment (rsarawgi) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ToXContentParams.java]: Added it back :)

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java (hunk 1 above)
Comment (rsarawgi) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java]: Update code to reflect this.

**Review by rsarawgi — COMMENTED** [review]
Thanks @droberts195 for taking a look. I've addressed the code review comments. Also fixed the CLA issue related to signature. The check passes now. Let me know if there's anything else that needs to be done. Thank you!

**Comment by droberts195** [comment]
Thanks @rsarawgi.  The Jenkins test I kicked off earlier found 3 lines that exceed the maximum permitted:

```
08:55:59 > Task :x-pack:plugin:ml:checkstyleTest FAILED
08:55:59 [ant:checkstyle] [ERROR] /dev/shm/elastic+elasticsearch+pull-request-2/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/JobResultsProviderIT.java:566: Line is longer than 140 characters (found 144). [LineLength]
08:55:59 [ant:checkstyle] [ERROR] /dev/shm/elastic+elasticsearch+pull-request-2/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/JobResultsProviderIT.java:609: Line is longer than 140 characters (found 144). [LineLength]
08:55:59 [ant:checkstyle] [ERROR] /dev/shm/elastic+elasticsearch+pull-request-2/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/JobResultsProviderIT.java:639: Line is longer than 140 characters (found 144). [LineLength]
```

(That's from https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/9104/console, which you can navigate to by clicking the red cross next to the third commit, which is the one that got tested.)

If you could split those 3 lines then I'll request another test run and if that passes then I'll integrate the change.

**Comment by rsarawgi** [comment]
@droberts195 updated the code.

**Comment by droberts195** [comment]
You've committed a massive chunk of commented out code in your 5th commit.  Like you accidentally pasted a whole file into a comment.

**Comment by rsarawgi** [comment]
Oops - Sorry about that! Reverted back my changes @droberts195

**Comment by droberts195** [comment]
Jenkins test this

**Comment by droberts195** [comment]
@rsarawgi please can you run `./gradlew :x-pack:plugin:core:precommit :x-pack:plugin:ml:precommit` from the top level of your local clone of the `elasticsearch` repo and fix the problems it reports.  This will avoid the situation where you push code that doesn't compile.  (You can also see the errors if you click the "Details" links next to the PR checks, but it's more efficient if you fix these basic problems locally before updating the PR.)

**Comment by rsarawgi** [comment]
Jenkins test this

**Comment by droberts195** [comment]
Jenkins test this

**Comment by rsarawgi** [comment]
@droberts195 Fixed the compilation and checkstyle issue -- but looks the build is still failing. Would need some help in figuring out why it's still failing

**Comment by droberts195** [comment]
Jenkins test this

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/bwc

**Comment by droberts195** [comment]
Jenkins run elasticsearch-ci/default-distro

**Comment by droberts195** [comment]
Jenkins test this

**Comment by droberts195** [comment]
> looks the build is still failing

@rsarawgi it was because since you created your PR branch hanging off master the version on the 7.x branch has been bumped from 7.5.0 to 7.6.0.  This was making the backwards compatibility tests fail.  I just merged master into your PR branch so hopefully the backwards compatibility tests will pass now.

**Comment by droberts195** [comment]
@rsarawgi the CI checks finally passed and I have merged the change.  Thanks for the contribution!

**Comment by rsarawgi** [comment]
Awesome, thanks @droberts195

**Pull request merged**